tests/test_get_column_info.py

```diff
@@ -52,37 +58,136 @@ def test_get_column_types(db_conn, table_schema_name, valid_table_name):
             "sgd1",
             "prepaid_card_ask_sample",
             start_partition=19000000,
             end_partition=19100000,
         ),
-        con=db_conn,
+        con=db_conn_string,
         coerce_float=False,
     )
     typed_df = convert_columns(df, db_columns, {})
     assert typed_df.schema.field_by_name("_block_number").type == pyarrow.uint32()
+
+
+def extract_to(output_folder, db_conn_string, db_conn=None, latest_block=None):
+    """Extracts the database contents out, optionally only up to a specific block
+
+    Args:
+        output_folder (AnyPath): The folder to write to
+        db_conn_string (str): the database connection string
+        db_conn (): the database connection, required for overriding the latest block
+        latest_block (int): override the latest block in the database
+    """
+    if latest_block is not None:
+        cur = db_conn.cursor()
+        cur.execute(f"UPDATE subgraphs.subgraph_deployment SET latest_ethereum_block_number={latest_block}")
+        db_conn.commit()
+        cur.close()
+    extract({
+            "name": "my_extract_name",
+            "version": "0.0.1",
+            "subgraph": "my_test_subgraph",
+            "tables": {
+                "prepaid_card_ask_sample": {
+                    "partition_sizes": [524288,32768,1024]
+                }
+            }
+        }, db_conn_string, output_folder)
+    return output_folder.joinpath("my_extract_name", "0.0.1")
+
+def test_write_out_results(db_conn_string):
+    
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir = AnyPath(temp_dir)
+        resulting_folder = extract_to(temp_dir, db_conn_string)
+        dataset = ds.dataset(resulting_folder.joinpath("data"), format="parquet")
+        df = dataset.to_table().to_pandas()
+        assert len(df) == 6
+        assert resulting_folder.joinpath("latest.yaml").exists()
+
+def test_writing_twice_when_block_increases_adds_data(db_conn_string, db_conn):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir = AnyPath(temp_dir)
+        resulting_folder = extract_to(temp_dir, db_conn_string, db_conn, 18888000)
+        dataset = ds.dataset(resulting_folder.joinpath("data"), format="parquet")
+        df = dataset.to_table().to_pandas()
+        assert len(df) == 2
+        assert resulting_folder.joinpath("latest.yaml").exists()
+    
+        # Move on some blocks and run again to check we're reading in the existing config and latest.yaml
+        resulting_folder = extract_to(temp_dir, db_conn_string, db_conn, 19000000)
+        dataset = ds.dataset(resulting_folder.joinpath("data"), format="parquet")
+        df = dataset.to_table().to_pandas()
+        # Partitions overlap so we can just remove the duplicates
+        df = df.drop_duplicates()
+        assert len(df) == 6
+        assert resulting_folder.joinpath("latest.yaml").exists()
+
+
+def test_second_run_fills_all_data_if_crashed_in_first(db_conn_string, db_conn):
```

Comment:
The description was really good when the example layed out. Maybe put that in here